third-party/realdds/include/realdds/dds-exceptions.h

```diff
@@ -0,0 +1,38 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <librealsense2/utilities/easylogging/easyloggingpp.h>
+#include <stdexcept>
+#include <string>
+
+
+// E.g.:
+//     DDS_THROW( runtime_error, "what happened" );
+//
+#define DDS_THROW( ERR_TYPE, WHAT )                                                                 \
+    throw realdds::dds_ ## ERR_TYPE ( WHAT )
+
+
+namespace realdds {
+
+
+class dds_runtime_error : public std::runtime_error
+{
+public:
+    dds_runtime_error( std::string const& str )
+        : std::runtime_error( str )
+    {
+        LOG_ERROR( "throwing: " << str);
+    }
+
+    dds_runtime_error( char const* lpsz )
```

Comment:
lpsz? long pointer size?

Reply:
long point to zero-terminated string.
Shows you my background (and maybe age?)...